kubeadm/KubeCluster.ps1

```diff
@@ -342,8 +342,16 @@ if ($Join.IsPresent)
     if ($Global:Cni -eq "flannel")
     {
         CreateExternalNetwork -NetworkMode $Global:NetworkMode -InterfaceName $Global:InterfaceName
+        WaitForNetworkInterface -IpAddress $Global:ManagementIp
         StartFlanneld 
-        WaitForNetwork $Global:NetworkName
+        
+        # Ensure Flannel created the HNS network
+        $flannelNetOk = WaitForNetwork -NetworkName $Global:NetworkName -DieOnFail $false
+        if (!$flannelNetOk)
+        {
+            RestartFlanneld
```

Comment:
yet, we are still forcing a flannel restart.
at this point and ideally, i want the assigned reviewers to take a look at this.

Reply:
makes sense... we can also cut that bit out.  In my latest tests the restart does not trigger, `WaitForNetworkInterface` takes care of the issue.  I just left this capability to restart FlannelD, as a last resort.